docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+If a block fails to upload then you need to re-upload the data for that block only|If a file fails to upload then you need to re-upload the complete file.
+
+### Multipart Upload Feature
+----------------------------
+
+Uploading parts of a file is done via the [multipart feature](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html); using this feature, each part of a file can be uploaded independently.
+After all parts of a file are uploaded successfully, Amazon S3 combines the parts as a single object.
+
+Please refer to the [Java code](http://docs.aws.amazon.com/AmazonS3/latest/dev/llJavaUploadFile.html) for uploading file into Amazon S3 bucket using multipart feature.
+
+### Module
+----------
+
+A **module** a group of operators pre-wired together so they work as a single conceptual entity in an application. Typically, a module will contain a set of input ports, output ports and configuration properties. The operators internal to the the module will be automatically configured based on the supplied module properties.
+
+### Operators in S3OutputModule
+-----------------------------------------
+
+Following diagram illustrates the DAG in this module:
+
+![](images/s3output/s3outputmodule.png)
+
+-   ***S3InitiateFileUploadOperator***
+    -   Initiate the upload for the file using **AmazonS3Client.initiateMultipartUpload(...)** method only if the number of blocks for a file is greater than 1. By successfully initiating the upload, S3 returns a response of type **InitiateMultipartUploadResult**, which includes the **upload ID**, which is the unique identifier for the multipart upload. This **upload ID** must be included in each operation like block upload and upload completion request..
+If the file has single block then the operator emits an empty string, this is an indication to downstream operators to not use the multi-part feature.
+    -   This operator emits the pair **(filemetadata, uploadId)** to **S3FileMerger** and the triple **(filePath, metadata, uploadId)** to **S3BlockUploadOperator**.
+
+-   ***S3BlockUploadOperator***
+    -   This operator upload the blocks into S3 using different calls which depend on number of blocks of a file.
+If the file has single block then upload the block omit using **AmazonS3Client.putObject(...)** call. S3 returns a response of type **PutObjectResult** which includes the **ETag**.
+If the file has more blocks then upload the block using **AmazonS3Client.uploadPart(...)** call. S3 returns a response of type UploadPartResult which includes the **ETag**. This **ETag** value must be included in the request to complete multipart upload.
```

Comment:
Please add backquotes around UploadPartResult as noted in my earlier review comment.